torch/distributed/_tensor/device_mesh.py

```diff
@@ -279,9 +298,11 @@ def scatter(
         # remove the check below once that is done.
         if output.is_meta:
             return None
-        dim_group = self._dim_groups[mesh_dim]
+        dim_group = self.get_dim_groups(mesh_dim)
+        assert isinstance(dim_group, ProcessGroup)
```

Comment:
yeah so it seems we need to do this everywhere after calling `device_mesh.get_dim_groups()`. We probably should avoid this.

Reply:
`get_dim_groups` now return either a ProcessGroup or a List of ProcessGroup depending on whether you pass a mesh dim or not, this assert is simply a type check to make sure mypy does not complain

Reply:
Yes I know. But I feel that having to add an assertion every time after calling `get_dim_groups` is not a good sign and we should avoid that by changing the return type.  Is there a reason why we want to return a List of `ProcessGroup`s instead of one PG that contains all the ranks within the list?

Reply:
i don't understand what you mean by "changing the return type". Can you clarify more on this? The reason why we want to return a list of process groups is that user want to have those groups for different mesh dimension, i.e. the 2-D parallel need this API to work with FSDP.

What do you mean by "instead of one PG that contains all the ranks within the list"?

Reply:
I propose that we separate this function into 2. One takes in a `dim` argument and returns a `ProcessGroup` (`get_group_by_dim`) and the other simply returns a `List[ProcessGroup]` (`get_dim_groups`), because adding assertion is quite not ideal.

With regard to "instead of one PG that contains all the ranks within the list", forget it. My mistake, I misunderstood the function's behavior.